app/controllers/settings_controller.py

```diff
@@ -577,6 +591,23 @@ def _update_model_from_view(self) -> None:
             self.settings_dialog.todds_overwrite_checkbox.isChecked()
         )
 
+        # themes tab
+        if self.settings_dialog.enable_themes_checkbox.isChecked():
+            self.settings.enable_themes = True
+        else:
+            self.settings.enable_themes = False
+        # get theme names
+        selected_theme = self.settings_dialog.themes_combobox.currentText()
+        self.settings.theme = selected_theme
+        available_themes = [folder.name for folder in Themes.get_available_themes()]
+
+        if selected_theme in available_themes:
+            self.settings.theme = selected_theme
+        else:
+            # Handle invalid theme selection
+            logger.warning(f"Invalid theme selected: {selected_theme}")
+            self.settings.theme = "RimPy"  # Set a default theme
```

Comment:
What if there is no theme RimPy for some reason?

Also, you are modifying the internally stored theme without modifying what is actually shown to the user. Its logged as a warning but it can be potentially confusing to debug for the user if the combo box says one thing, but the theme stays as RimPy.

Reply:
Done